packages/builder/src/schemas.zod.ts

```diff
@@ -0,0 +1,304 @@
+import { z } from 'zod';
+import { handleZodErrors } from './error/zod';
+
+/// ================================ INPUT CONFIG SCHEMAS ================================ \\\
+
+export const contractSchema = z
+  .object({
+    artifact: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        create2: z.boolean(),
+        from: z.string(),
+        nonce: z.string(),
+        abi: z.string(),
+        abiOf: z.array(z.string()),
+        args: z.array(z.any()),
+        libraries: z.record(z.string()),
+
+        // used to force new copy of a contract (not actually used)
+        salt: z.string(),
+
+        value: z.string(),
+        overrides: z.object({
+          gasLimit: z.string(),
+        }),
+
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const importSchema = z
+  .object({
+    source: z.string(),
+  })
+  .merge(
+    z
+      .object({
+        chainId: z.number().int(),
+        preset: z.string(),
+        depends: z.array(z.string()),
+      })
+      .deepPartial()
+  );
+
+export const invokeSchema = z
+  .object({
+    target: z.array(z.string()).nonempty(),
```

Comment:
validate that its either an address or valid cannon step name (cannon step names can only have certain characters I believe?)

Reply:
I can validate that its either an address or interpolated value, if we start to validate "known" step names it gets tricky since new plugins can add new step names